public/hist_nd.pro

```diff
@@ -131,13 +133,25 @@
 ;
 ;##############################################################################
 
-function hist_nd,V,bs,MIN=mn,MAX=mx,NBINS=nbins,REVERSE_INDICES=ri
+function hist_nd,V,bs,MIN=mn,MAX=mx,NBINS=nbins,REVERSE_INDICES=ri,NAN=nan
+
+  nan=(n_elements(nan) ne 0) && keyword_set(nan)
+  if (~nan && ~array_equal(finite(V), 1)) then begin
+    message, 'NaN detected; setting NAN keyword', /continue
```

Comment:
I don't know if you really need to send this message. The "IDL way" is to not be too chatty. Either just do the right thing quietly or throw an error.

Reply:
A similar thing is done in [cghistogram.pro](https://github.com/idl-coyote/coyote/blob/ff720efbfa9a5070c0e9b772d1bcedbe718791dc/cghistogram.pro#L194-L196). I agree with you about dropping the message, but I was trying to be consistent.